[designate] seed generic domains using global.domains_seeds

wagnerd3 opened the pull request:
In an attempt to make deployment of new domains easier, @majewsky recently introduced `global.domain_seeds.customer_domains`. As this could replace the old `skip_hcm_domain`, poses the opportunity to distinguish between internal and external domains in future and accommodate other specialties we would like to suggest this on seeds of other services too. 

The expressions are technically equal, only the `ora` domain was added. If this is on purpose, I can introduce an exception. I excepted `iaas-` (external) domains already, as I thought they are probably missing on purpose, but I could also add those if desired. 
I ran `h3 diff` against `qa-de-1`, you can have a look at the output here: [designatediff.txt](https://github.com/user-attachments/files/19233309/designatediff.txt)

galkindmitrii commented:
Hi @wagnerd3, thanks for working on that.

The `.Values.global_setup` is a special case and separate region where customer domains should not be present at all (any).
Please adjust accordingly, that `{{- if .Values.global_setup }}` we only seed default domains (`ccadmin`, `cc3test` etc.) + `domain-global-seed`.

Thank you.

galkindmitrii reviewed (changes requested):
^^

wagnerd3 commented:
@galkindmitrii I updated the PR and now implemented the global region also via the `domain_seeds.customer_domains` variable. In this region we now have `domain_seeds.customer_domains = list ("global")` coming from the values. 
Additionally, I added all domains (even `ccadmin, cc3test`) to the loop except `default, tempest` - because they are completely different from the rest. 

I generated some more `helm diff` outputs to check (this time in the `dyff` syntax for better overview):
[designateGlobalQA1diff.txt](https://github.com/user-attachments/files/19348707/designateGlobalQA1diff.txt)
[designateGlobalDE1diff.txt](https://github.com/user-attachments/files/19348710/designateGlobalDE1diff.txt)
[designateQA1diff.txt](https://github.com/user-attachments/files/19348711/designateQA1diff.txt)
[designateDE1diff.txt](https://github.com/user-attachments/files/19348716/designateDE1diff.txt)

You can see, that a) only the orders change and b) as already asked initially the `ora` domain gets some missing assignments in the non-global regions. Let me know if this gives enough confidence now to merge. :-)

wagnerd3 commented:
@galkindmitrii I saw some errors from the designate nanny seed in `global` region where the `monsoon3global/domain-default-seed` is required. As designate is set up in some form of HA-setup, it is currently also seeded from all the involved regions. The seeding for global is not layed out in such distributed fashion though, because the seedings are one-time-actions and not required to work when one region is down. The domain seeds are just seeded from eu-de-2. I would include some lines in this PR to limit this for global, is this okay with you?

wagnerd3 commented:
I updated the PR with regard to the above.

galkindmitrii reviewed (changes requested):
Unfortunately changing `.Values.global_setup` to `.Values.global.is_global_region ` introduces many issues:
1) All other chart [templates need to be updated](https://github.com/search?q=repo%3Asapcc%2Fhelm-charts%20global_setup&type=code)
2) Secrets config need to be updated, services redeployed
3) Keystone which uses the same variable/approach will then differ.

Therefore, I'd not touch that `global_setup` setting.

majewsky commented:
To clarify, the problem with `.Values.global_setup` is that it falls out of the standard pattern of "values shared by multiple services are declared in `$REGION/values/globals.yaml` in the secrets repo and are declared under `.Values.global`. The way that `.Values.global_setup` is used makes it hard for us to adopt in our own services, at least because its semantics are not properly documented, so we are proposing a standardized value that follows the usual semantics.

> All other chart templates need to be updated.

Yes, we plan to do that.

> Secrets config need to be updated

Already done (see `global/values/globals.yaml` in the secrets repo).

> services redeployed

Well yes, but that happens frequently anyway. We are not in a hurry with this, it does not block anything else if you do not do a deployment immediately.

> Keystone which uses the same variable/approach will then differ.

Can you explain this point further, please?

galkindmitrii commented:
> Can you explain this point further, please?

https://github.com/search?q=repo%3Asapcc%2Fhelm-charts+path%3A%2F%5Eopenstack%5C%2Fkeystone%5C%2F%2F+global_setup&type=code

wagnerd3 commented:
> > Can you explain this point further, please?
> 
> https://github.com/search?q=repo%3Asapcc%2Fhelm-charts+path%3A%2F%5Eopenstack%5C%2Fkeystone%5C%2F%2F+global_setup&type=code

I would soon open PRs for the other references of the old variable. I think the cut-over time in the repo's itself will be 1-2 weeks only. I will then just wait for all deployments before I delete the old variable in secrets.

galkindmitrii commented:
So there are [55 files where global_setup](https://github.com/search?q=repo%3Asapcc%2Fhelm-charts%20global_setup&type=code) is currently used in `helm-charts` and I'm not feeling very comfortable having partial migration to the new `is_global_region`.

Maybe @braunmatt or @mutax sees it different...

majewsky commented:
So if we make a PR that adjusts Designate and Keystone and domain-seeds all at once, that would be fine?

galkindmitrii commented:
> So if we make a PR that adjusts Designate and Keystone and domain-seeds all at once, that would be fine?

Please include Designate and Designate Nanny into 1 PR. Keystone separately since there are other approvals.

Thx.

galkindmitrii reviewed (approved):
lgtm
